### ISSUE TITLE
todo-keywords sorting does not work

### ISSUE THREAD

**Issue by thorbenandresen** [alphapapa/org-ql#242]
I cannot get `org-ql-search` to sort my keywords by the `org-todo-keywords-1` sequence (as previously discussed in #217)

Here is my minimal repro case: 
```org
#+TITLE: Org Ql Test
#+TODO: URGENT NEXT P2 P3 TODO WAIT | DONE
* P2 Item
* TODO Item
* NEXT Item
* WAIT Item
* P3 Item
* URGENT Item
* P2 Item
```

```elisp
(defun ta/org-ql/test ()
  (interactive)
  (org-ql-search "~/Dropbox/org/gtd/org-ql-test.org"
    '(todo)
    :sort '(todo)
    :title "Test")
  )
```

The screenshot below shows the generated org-ql buffer with the incorrect sorting, as well as the source org buffer and the value of `org-todo-keywords-1` of the org buffer.

![screen shot 2021-08-10 at 09 37 15 PM](https://user-images.githubusercontent.com/8238976/128925270-8e29a373-e8eb-4b92-9567-d3528a45bd9b.jpg)

org-ql 0.5.2
emacs 27.2
doom 3.0.0-alpha HEAD 
MacOS 11.4 Darwin 20.5.0 arm64 ns

**Comment by alphapapa** [comment]
> org-ql 0.5.2

Please test with the latest version of this package.

**Comment by thorbenandresen** [comment]
The issue is present on the latest version

```
    Package: org-ql
     Source: Straight
     Pinned: unpinned
      Build: grafted, HEAD -> master, tag: list, origin/master, origin/HEAD 94f9e6f 2021-07-12 21:33:24 -0500
             Build location: ~/.emacs.d/.local/straight/build-27.2/org-ql/
             Repo location: ~/.emacs.d/.local/straight/repos/org-ql/
     Recipe: (:type git :flavor melpa :files
                    (:defaults
                     (:exclude "helm-org-ql.el")
                     "org-ql-pkg.el")
                    :host github :repo "alphapapa/org-ql" :package "org-ql" :local-repo "org-ql")
                Homepage: https://github.com/alphapapa/org-ql
    Modules: Declared by the following Doom modules:
             :private packages (no readme)
             :private modules (no readme)
    Configs: This package is not configured anywhere
```

**Comment by alphapapa** [comment]
Ok, thanks.  I'll take a look, but I probably won't get to this for a while, as I have other priorities at the moment.  Your test case should make solving it easier, though; feel free to attempt a fix in the meantime.  :)

**Comment by thorbenandresen** [comment]
Thank you and no worries, I will also be on vacation / AFK from tomorrow. I will  try to fix it afterwards, where you ever able to reproduce my test case?

**Comment by alphapapa** [comment]
Yes, and I think I found the problem, an oversight on my part: `org-ql--sort-by` is not *always* called in each item's buffer (but it could be, depending on where the chain of execution began), so the value of `org-todo-keywords-1` may be nil when finding the position of keywords in it.

I'm not sure of the best way to solve that, because at this point in the code, the items being sorted are not necessarily from a single buffer, so the value of `org-todo-keywords-1` wouldn't necessarily be the same for each item, anyyway.  I suppose the sorting could be done for each buffer's items, but then it would need to be done again for the whole set, depending on which sorters are being used.  

It's not especially straightforward.  Unless there's a really simple solution that I'm overlooking, I probably wouldn't try to fix this before tagging 0.6.
